rpgsaga/computer/src/main.ts

```diff
@@ -0,0 +1,42 @@
+export class Computer {
+    private _name : string;
+    energy_consumption : number;
+    constructor(deviceName: string, device_energy_consumption : number) {
+        this._name = deviceName;
+        this.energy_consumption = device_energy_consumption;
+    }
+    print_name(): void {
+        console.log(`Имя: ${this._name}`);
+    }
+    public get name(){
+        return this._name
+    }
+    public set name(input : string){
+        this._name = input;
+    }
+}
+export class HDD extends Computer {
+    private _memory : number;
+    data_transfer_time_sec : number;
+    constructor(deviceName : string, memory : number, data_transfer_time_sec : number, device_energy_consumption : number) {
+        super(deviceName, device_energy_consumption);
+        this._memory = memory;

```

Comment:
```suggestion
        this.memory = memory;

```
Это приведет к вызову сеттера с соответствующей проверкой